NitroxClient/GameLogic/InitialSync/StoryGoalInitialSyncProcessor.cs

```diff
@@ -90,11 +89,34 @@ private static IEnumerator SetupTrackers(InitialPlayerSync packet)
             }
         }
         techTypesToRemove.ForEach(techType => storyGoalManager.itemGoalTracker.goals.Remove(techType));
-        yield break;
+
+        // OnGoalUnlock might trigger the creation of a signal which is later on set to invisible when getting close to it
+        // the invisibility is managed by PingInstance_Set_Patches and is restored during PlayerPreferencesInitialSyncProcessor
+        // So we still need to recreate the signals at every game launch
+
+        // To avoid having the SignalPing play its sound we just make its notification null while triggering it
+        // (the sound is something like "coordinates added to the gps" or something)
+        PDANotification pdaNotification = onGoalUnlockTracker.signalPrefab.GetComponent<SignalPing>().vo;
+        onGoalUnlockTracker.signalPrefab.GetComponent<SignalPing>().vo = null;
+
+        foreach (OnGoalUnlock onGoalUnlock in onGoalUnlockTracker.unlockData.onGoalUnlocks)
+        {
+            if (completedGoals.Contains(onGoalUnlock.goal))
+            {
+                // Code adapted from OnGoalUnlock.Trigger
+                foreach (UnlockSignalData unlockSignalData in onGoalUnlock.signals)
+                {
+                    unlockSignalData.Trigger(onGoalUnlockTracker);
+                }
+            }
+        }
```

Comment:
If we suppress the OnSceneObjectsLoaded, and call onGoalUnlockTracker&coumpoundTracker init by hand, shouldn't it follow the traditional way the game is using to restore his story data ?